types/react-redux/index.d.ts

```diff
@@ -50,73 +65,73 @@ interface ComponentMergeDecorator<TMergedProps, TOwnProps> {
  * @param mergeProps
  * @param options
  */
-export declare function connect(): ComponentDecoratorInfer<DispatchProp<any>>;
+export declare function connect(): InferableComponentEnhancer<DispatchProp<any>>;
 
 export declare function connect<TStateProps, no_dispatch, TOwnProps>(
     mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>
-): ComponentDecorator<TStateProps, TOwnProps>;
+): InferableComponentEnhancerWithProps<TStateProps & DispatchProp<any>, TOwnProps>;
```

Comment:
That's how `react-redux` works:

> If you do not supply your own mapDispatchToProps function or object full of action creators, the default mapDispatchToProps implementation just injects dispatch into your component’s props.

https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options